ister_gui.py

```diff
@@ -1223,19 +1223,31 @@ def _reset_network(self, _):
         except FileNotFoundError:
             pass
 
-        try:
-            subprocess.call(['/usr/bin/systemctl', 'restart',
-                             'systemd-networkd', 'systemd-resolved'])
-        except:
-            raise urwid.ExitMainLoop()
+        self._restart_networkd_resolved()
 
-        # give the network a chance to start up again
-        time.sleep(.1)
+        # set necessary flags, then reset static edit text.
         self.static_ready = False
         self.reset = True
         self.set_static_edits()
         raise urwid.ExitMainLoop()
 
+    def _restart_networkd_resolved(self):
+        """Restart the network services then poll systemctl status output until
+        both are back up"""
+        try:
+            subprocess.call(['/usr/bin/systemctl', 'restart',
+                             'systemd-networkd', 'systemd-resolved'])
+            while True:
```

Comment:
sleep 1 is an awfully long time to sleep though. I wasn't running into any issues with sleeping a tenth of a second.